### DISCUSSION TITLE
Does jreleaser support auto-versioning?

### DISCUSSION THREAD

**Discussion by quintesse** [jreleaser/jreleaser#1687] [Q&A] [answered]
I wondered if jreleaser supports something that works similarly to the maven-release-plugin where it will automatically increase the project version (and the scm.tag), commit that, tag it and finally increase the version again but with SNAPSHOT (and HEAD for scm.tag)?

**Comment by aalmiray** [comment, marked as answer]
No, not at the moment as the version number is considered as an input. Also, it may seem straight forward to perform these changes for a Maven based project but given that JReleaser supports any kind of project regardless of its chosen build tool it would be nearly impossible to offer this behavior in a build tool agnostic way.

**Reply by quintesse** [reply]
Understandable 👍 
And a way that, when using maven, it takes the version from the pom.xml file?
It's perhaps too small an advantage but it would remove one duplication of the version number from the equation.

**Reply by aalmiray** [reply]
AFAICT the jreleaser-maven-plugin already takes that info into account https://jreleaser.org/guide/latest/tools/jreleaser-maven.html#_defaults

**Reply by quintesse** [reply]
Ok thanks! I'll take a look at that, so far I've been using the command line tool. But it makes sense that the maven plugin would take care of that.

Let me sneak in another question (or if you'd like I can open a separate discussion for it): is there a way to have the assemble step be performed automatically as part of the release step?

**Reply by aalmiray** [reply]
Yes, it is possible if you use a command or script hook to invoke the CLI as hinted by the SLSA java builder 

https://github.com/jreleaser/release-action/tree/java?tab=readme-ov-file#slsa-builder

However, now that I think more about it there's a high chance assembled distros and artifacts won't be ready at the right time during configuration. I'm afraid hooks won't help. 

There's a proposal to configure a custom workflow https://github.com/jreleaser/jreleaser/issues/1675

**Reply by quintesse** [reply]
This is now my release process: https://github.com/codejive/java-jpm/blob/main/RELEASE.md, it's getting there, but it still isa bit more involved than `mvn -B release:clean release:prepare release:perform`. Of course jreleaser does a gazillion things more, but still, I wouldn't mind at all getting to a simpler process (if only to prevent me from making mistakes 😉 )
